packages/terra-form-select/src/native-select/NativeSelect.jsx

```diff
@@ -0,0 +1,247 @@
+  const value = placeholder.value ? placeholder.value : defaultPlaceholderValue;
+  const attrs = placeholder.allowClear ? {} : { disabled: true, hidden: true };
+  return <option value={value} {...attrs}>{display}</option>;
+};
+
+const createOptions = options => {
+  const currentOptGroupKeys = [];
+
+  return options.map(current => {
+    if (current.options) {
+      const optGroupKeyIndex = geOptGroupKeyIndex(current.display, currentOptGroupKeys);
+      const optGroupKey = getOptGroupKey(current.display, optGroupKeyIndex);
+      currentOptGroupKeys.push(optGroupKey);
+
+      return (
+        <optgroup key={optGroupKey} label={current.display}>
+          {createOptions(current.options)}
+        </optgroup>
+      );
+    }
+    return <option key={`${current.value}`} value={current.value}>{current.display}</option>;
+  });
+};
+
+const NativeSelect = ({
+  ariaDescribedBy,
+  ariaLabel,
+  attrs,
+  disabled,
+  defaultValue,
+  id,
+  intl,
+  isInvalid,
+  isIncomplete,
+  onChange,
+  options,
+  placeholder,
+  required,
+  value,
+  ...customProps
+}) => {
+  const [uncontrolledValue, setUncontrolledValue] = useState(!value ? defaultValue || getFirstValue(options, placeholder) : undefined);
+  const refIsControlled = useRef(!uncontrolledValue);
+  const refSelect = useRef();
+  const theme = React.useContext(ThemeContext);
+
+  const handleOnMouseDown = () => {
+    refSelect.current.setAttribute('data-focus-interaction', 'mouse');
+  };
+  const handleOnBlur = () => {
+    refSelect.current.setAttribute('data-focus-interaction', 'none');
+  };
+  const handleOnFocus = () => {
+    if (refSelect.current.getAttribute('data-focus-interaction') !== 'mouse') {
+      refSelect.current.setAttribute('data-focus-interaction', 'keyboard');
+    }
+  };
+  const handleOnChange = event => {
+    if (onChange) {
+      onChange(event);
+    }
+    if (!refIsControlled.current) {
+      setUncontrolledValue(event.currentTarget.value);
+    }
+  };
+
+  let currentValue = refIsControlled.current ? value : uncontrolledValue;
+  let currentDisplay = getDisplay(currentValue, options, placeholder, intl);
+  if (!currentDisplay) {
+    currentValue = getFirstValue(options, placeholder);
+    currentDisplay = getDisplay(currentValue, options, placeholder, intl);
+  }
+
+  const selectAttrs = {
+    'aria-describedby': ariaDescribedBy,
+    'aria-label': ariaLabel,
+    id,
+    disabled,
+    'aria-invalid': isInvalid || undefined,
+    required,
+    value: currentValue,
+  };
+
+  const nativeClassNames = classNames(
+    cx(
+      'native',
+      theme.className,
+      { disabled },
+      { invalid: isInvalid },
+      { incomplete: required && isIncomplete },
+      { placeholder: isCurrentPlaceholder(selectAttrs.value, placeholder) },
+    ),
+    customProps.className,
+  );
+
+  return (
+    <div
+      {...customProps}
+      className={nativeClassNames}
+      ref={refSelect}
```

Comment:
Do we need to expose this ref for consumers? Or for the `select` element?

Reply:
Updated to use refCallback.